script/formatting/validator.py

```diff
@@ -0,0 +1,178 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ======================================================
+# GOAL : Warn about print statements and commented code
+# ======================================================
+
+
+import argparse
+import os
+import re
+import functools
+
+from formatter import LOG
+
+# ==============================================
+# LOGGING CONFIGURATION
+# ==============================================
+
+EXIT_SUCCESS = 0
+EXIT_FAILURE = -1
+
+VALIDATOR_PATTERNS = [ re.compile(patterns) for patterns in [
+    r"print"
+    ]
+]
+
+CODE_SOURCE_DIR = os.path.abspath(os.path.dirname(__file__))
+EVA_DIR = functools.reduce(os.path.join,
+                           [CODE_SOURCE_DIR, os.path.pardir, os.path.pardir])
+
+EVA_SRC_DIR = os.path.join(EVA_DIR, "src")
+EVA_TEST_DIR = os.path.join(EVA_DIR, "test")
+EVA_SCRIPT_DIR = os.path.join(EVA_DIR, "script")
+
+DIRS = [EVA_SRC_DIR,EVA_TEST_DIR]
+
+# ==============================================
+# UTILITY FUNCTION DEFINITIONS
+# ==============================================
+
+def comment_contains_code(line):
+    line = line.lstrip()
+    
+    if 'utf-8' in line:
+        return False
+
+    if not line.startswith('#'):
+        return False
+
+    line = line.lstrip(' \t\v\n#').strip()
+
+    regex_list = [
+        r"def .+\)[\s]*[->]*[\s]*[a-zA-Z_]*[a-zA-Z0-9_]*:$",
```

Comment:
Add a comment discussing the need for these regexes?

Reply:
Added